File: api/Models/SaleDetail.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    public class SaleDetail
    {
        public int SaleId { get; set; }
        public Sale Sale { get; set; } = default!;
        public int ProductId { get; set; }
        public Product Product { get; set; } = default!;
        [Required]
        public int Quantity { get; set; }
        [Required]
        [Precision(18, 2)]
        public decimal UnitPrice { get; set; }
        [Required]
        [Precision(18, 2)]
        public decimal Subtotal => Quantity * UnitPrice;
    }
}

File: api/app/models/sale_detail.ts
import { BaseModel, belongsTo, column } from '@adonisjs/lucid/orm';
import Sale from './sale.js';
import Product from './product.js';
import type { BelongsTo } from '@adonisjs/lucid/types/relations';

export default class SaleDetail extends BaseModel {
  @column()
  declare saleId: number;

  @column()
  declare productId: number;

  @column()
  declare quantity: number;

  @column()
  declare unitPrice: number;

  @belongsTo(() => Sale)
  declare sale: BelongsTo<typeof Sale>;

  @belongsTo(() => Product)
  declare product: BelongsTo<typeof Product>;
}
